main/src/ca/uwaterloo/flix/language/phase/unification/UnificationCache.scala

```diff
@@ -15,13 +15,15 @@
  */
 package ca.uwaterloo.flix.language.phase.unification
 
+import org.sosy_lab.pjbdd.api.DD
 import java.util.concurrent.ConcurrentHashMap
 
 object UnificationCache {
   /**
     * A Global (per-JVM) cache of unification queries.
     */
-  val Global: UnificationCache[BoolFormula] = new UnificationCache()
+  val GlobalBdd: UnificationCache[DD] = new UnificationCache()
+  val GlobalBool: UnificationCache[BoolFormula] = new UnificationCache()
 }
 
```

Comment:
But thinking about it, what's the usefulness of having the DD cache?

That we avoid a translation into BoolFormula?

Reply:
I'm not sure what you mean by this. Since we switched to the version where for each call to unify() we only use BDDs or BoolFormulas, this is just the equivalent of the BoolFormula cache, but for larger instances

Reply:
here is what I think I mean:

If you have a 5 variable formula, then I think its translated into a BDD, then we lookup the substitution (which is BDD based!) and translate that back into a type.

Reply:
Thus, unless I am confused, we have to go from BDD subst -> explicit/type subst on every cache hit.

Reply:
Yes, but if we do not use the cache then we have to do the whole SVE computation with BDDs, get the same BDD subst and then translate it to a type subst anyway?
Unless you want the cache to work on types instead of BoolFormulas/BDDs? Then we could skip the converting steps

Reply:
I would have the cache on BoolFormulas.

But now I wonder if we should even have a cache, since the formulas are large, and less likely to be hit?

Reply:
We could try running with and without the BDD cache to see if it effects performance?

Reply:
How many times is it even hit?

Reply:
How about you fix the other issues I mentioned, then we do a merge, and we can do more experiments.